packages/api/src/middleware/acl/checkOwnership.js

```diff
@@ -1,73 +1,73 @@
+//  *  it under the terms of the GNU General Public License as published by
+//  *  the Free Software Foundation, either version 3 of the License, or
+//  *  (at your option) any later version.
+//  *
+//  *  LiteFarm is distributed in the hope that it will be useful,
+//  *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+//  */
 
-const { Model } = require('objection');
+// const { Model } = require('objection');
 
-// relation must be one of field, plan, farmExpense, farmCrop, or to do
-const checkOwnership = (relation) => {
-  return async function authRelation(req, res, next) {
-    const knex = Model.knex();
-    const id = req.params.id;
-    if (id) {
-      let sqlQuery;
-      // console.log(farm_id);
-      // user id is contained in attribute sub in this format: 'auth0|5b0560215d7d1617fd7ed217'
-      const user_id = req.user.user_id
-      // console.log(`check ${relation}_id`, user_id, id);
+// // relation must be one of field, plan, farmExpense, farmCrop, or to do
+// const checkOwnership = (relation) => {
+//   return async function authRelation(req, res, next) {
+//     const knex = Model.knex();
+//     const id = req.params.id;
+//     if (id) {
+//       let sqlQuery;
+//       // console.log(farm_id);
+//       // user id is contained in attribute sub in this format: 'auth0|5b0560215d7d1617fd7ed217'
+//       const user_id = req.user.user_id
+//       // console.log(`check ${relation}_id`, user_id, id);
 
-      switch (relation) {
-      case 'shift':
-      case 'log':
-      case 'notification':
-        sqlQuery = `SELECT DISTINCT u.user_id
-        FROM "${relation}" r, "users" u
-        WHERE u.user_id = '${user_id}' AND r.${relation}_id = '${id}' AND u.user_id = r.user_id`;
-        break;
-      case 'sale':
-        sqlQuery = `SELECT DISTINCT u.user_id
-        FROM "${relation}" s, "users" u
-        WHERE u.user_id = '${user_id}' AND s.${relation}_id = '${id}' AND u.farm_id = s.farm_id`;
-        break;
-      case 'managementPlan':
-        sqlQuery = `SELECT DISTINCT u.user_id
-        FROM "${relation}" r, "users" u, "field" f
-        WHERE u.user_id = '${user_id}' AND r.management_plan_id = '${id}' AND f.field_id = r.field_id AND f.farm_id = u.farm_id`;
-        break;
-      case 'nitrogenSchedule':
-        sqlQuery = `SELECT DISTINCT u.user_id
-        FROM "${relation}" r, "users" u
-        WHERE u.user_id = '${user_id}' AND r.nitrogen_schedule_id = '${id}' AND u.farm_id = r.farm_id`;
-        break;
-        //TODO potential bug
-      default:
-        sqlQuery = `SELECT DISTINCT u.user_id
-        FROM ? r, "users" u
-        WHERE u.user_id = ? AND r.?_id = ? AND u.farm_id = r.farm_id`;
-      }
+//       switch (relation) {
+//       case 'shift':
+//       case 'log':
+//       case 'notification':
```

Comment:
It looks like there is something occuring here to do with notifications - are we sure this is not in use currently?

Reply:
Good question, but I'm sure it's unrelated to notifications as we currently know them. There were some references to "notifications" in the code/db before the recent feature development, but they were obsolete/unused prework for a somewhat different approach to the concept. There's still a bit of confusing stuff in the navbar-- one of the other icons was already referred to as "notifications" for some reason.